src/Router.ts

```diff
@@ -201,13 +250,15 @@ export class Router<C extends Context> extends Evented {
 	private _history?: History;
 	private _routes: Route<Context, Parameters>[];
 	private _started?: boolean;
+	private _outletStack: Map<string, any> = new Map<string, OutletContext>();
+	private _outletRouteMap: Map<string, Route<any, any>> = new Map<string, Route<any, any>>();
```

Comment:
Looking more at the code, this is really `_outletKeyRouteMap` but you _could_ use a weak map here instead.